modules/tinymce/src/plugins/advlist/main/ts/ui/Buttons.ts

```diff
@@ -43,9 +43,22 @@ const isWithinList = (editor: Editor, e: EditorEvent<NodeChangeEvent>, nodeName:
   return lists.length > 0 && lists[0].nodeName === nodeName;
 };
 
+const isEditableSelection = (editor: Editor, e: EditorEvent<NodeChangeEvent>): boolean => {
+  const root = editor.getBody();
+  let parent: HTMLElement | null = e.element as HTMLElement;
+  while (parent !== root && parent) {
+    if (editor.dom.getContentEditable(parent) === 'false') {
+      return false;
+    }
+    parent = parent.parentElement;
+  }
+  return true;
```

Comment:
This smells like `PredicateExists.ancestor()` but I'm not sure swapping it to use sugar nodes is worth the slight reduction in code. It is a _very_ common pattern, though.

There's sugar `ContentEditable.isEditable`, `editor.dom.getContentEditableParent`, probably others I'm forgetting, surely one of them is usable for this 🤔

Reply:
Yea that was the takeaway from @lnewson where this is reused within the lists plugin its only used once in Advlist. This function check all contenteditable parents and returns false if there is one false element, not only finding the first parent. This is because getContentEditableParent will be true when editing a contenteditable true list element which is within a contenteditable false list.